documentation.md

```diff
@@ -0,0 +1,117 @@
+
+- This [guide](https://albumentations.ai/docs/examples/pytorch_classification/) is extremely helpful for our classification task. You develop your own class that inherits from the torch dataset. To see an example of implementation for our dataset, please check this [notebook](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/auto_exp1/notebooks/autoalbument.ipynb).
+- Some transformations that believed to  be helpful - [here] (https://github.com/albumentations-team/albumentations#pixel-level-transforms) is the entire list:
+- CLAHE
+- Downscale
+- Equalize 
+- Hue Saturation (may not apply if no color channels)
+- ISO noise 
+- Invert Image
+- Random Birghtness
+- Random Contrat 
+- Random Snow
+- Shift Scale rotate 
+
+To use the [autoalbument](https://albumentations.ai/docs/autoalbument/) :
+
+1. First run the following command:
+`autoalbument-create --config-dir ~/mds/capstone/capstone-gdrl-lipo/experiments --task classification --num-classes 2` This creates all the files needed for you.
+
+2.  Next, you need to implement your own [torch custom dataset](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/auto_exp1/experiments/dataset.py). Next you set up the [search.yaml ](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/auto_exp1/experiments/search.yaml) file to ensure what it needs to search for (you can add preprocessing steps that apply to every image here, choose the batch size, number of workers, type of model, device available (cuda or cpu) and number of epochs). 
+
+3. Next, run the following command:
+`autoalbument-search --config-dir ~/mds/capstone/capstone-gdrl-lipo/experiments`
+
+4. Finally, you get a [json file](https://github.com/UBC-MDS/capstone-gdrl-lipo/tree/auto_exp1/experiments/outputs/2021-05-11/16-45-11/policy)  that can them be loaded in and applied to your transformation. See the transformation exploration [here](https://github.com/UBC-MDS/capstone-gdrl-lipo/blob/auto_exp1/notebooks/autoalbument.ipynb).
+
+### Loss Functions
+
+A loss function, also known as cost function or error function, is a function that maps a set of parameter values for the network onto a scalar value. The scalar value is an indication of how well the parameters are comleting their assigned tasks. In optimization problems, we seek to minimize the loss function. 
+
+The following functions were considered for the choice of the loss function used in the convolutional neural network model:
+
+1. BCEWithLogitsLoss
+2. NLLLoss
+3. Sum of Log Loss
+4. Hinge
+5. The mean absolute error 
+
+- BCEWithLogitsLoss: Cross-Entropy loss [nn.BCEWithLogitsLoss](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html#torch.nn.BCEWithLogitsLoss) combines a Sigmoid layer and the BCELoss in one single class. This loss function in often used in classification problems as a measure of reconstruction error. It uses log loss, log loss for logistic regression is a special case for cross-entropy loss for multi-class classification.
```

Comment:
Suggestion: BCEWithLogitsLoss: Cross-Entropy loss [nn.BCEWithLogitsLoss](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html#torch.nn.BCEWithLogitsLoss) combines a Sigmoid layer and the BCELoss in one single class. This loss function in often used in classification problems as a measure of reconstruction error. It uses log loss, which for logistic regression is a special case for cross-entropy loss for multi-class classification.